src/novmm/machine/ram.go

```diff
@@ -16,18 +16,24 @@ package machine
 
 import (
 	"math"
+	"sync"
 	"unsafe"
 )
 
 type Ram struct {
-	Data []byte `json:"data"`
+	Data   []byte `json:"data"`
+	rwlock sync.RWMutex
```

Comment:
This could just be a regular Mutex. The operations are so short that the RWMutex is almost certainly overkill (since it's two locks and a waitlist).